execution/executor/src/block_processor.rs

```diff
@@ -43,23 +40,32 @@ enum Mode {
     Syncing,
 }
 
+fn error_when_syncing(id: HashValue) -> Error {
+    let message = format!("Syncing. Unable to serve request for block {:x}.", id);
+    warn!("{}", message);
+    format_err!("{}", message)
+}
+
 pub(crate) struct BlockProcessor<V> {
     /// Where the processor receives commands.
     command_receiver: mpsc::Receiver<Command>,
 
     /// The timestamp of the last committed ledger info.
     committed_timestamp_usecs: u64,
 
-    committed_trees: ExecutedTrees,
+    committed_trees: Arc<Mutex<ExecutedTrees>>,
 
     /// The main block tree data structure that holds all the uncommitted blocks in memory.
-    block_tree: BlockTree<TransactionBlock>,
+    blocks_to_execute: VecDeque<(
+        ExecutableBlock,
+        oneshot::Sender<Result<(ProcessedVMOutput, StateComputeResult)>>,
+    )>,
 
     /// The blocks that are ready to be sent to storage. After pruning `block_tree` we always put
     /// the blocks here before sending them to storage, so in the case when storage is temporarily
     /// unavailable, we will still prune `block_tree` as normal but blocks will stay here for a bit
     /// longer.
-    blocks_to_store: VecDeque<TransactionBlock>,
+    block_batch_to_commit: Option<(CommittableBlockBatch, oneshot::Sender<Result<()>>)>,
```

Comment:
same here. I think we no longer need this thing, if this code becomes pretty much stateless.